homeassistant/components/tedee/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Tedee sensor entities."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pytedee_async import TedeeLock
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import TedeeDescriptionEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class TedeeSensorEntityDescription(SensorEntityDescription):
+    """Describes Tedee sensor entity."""
+
+    value_fn: Callable[[TedeeLock], int | None]
+
+
+ENTITIES: tuple[TedeeSensorEntityDescription, ...] = (
+    TedeeSensorEntityDescription(
+        key="battery_sensor",
+        translation_key="battery_sensor",
```

Comment:
If there is a device class and no translation key or name it can take the device class translation, so if you remove translation_key, it will work exactly as now, without creating more translation keys